transports/tcp/src/lib.rs

```diff
@@ -1329,4 +1328,35 @@ mod tests {
             assert!(rt.block_on(cycle_listeners::<tokio::Tcp>()));
         }
     }
+
+    #[test]
+    fn test_listens_ipv4_ipv6_separately() {
+        fn test<T: Provider>(port: u16) {
+            let mut tcp = Transport::<T>::default().boxed();
+            let listener_id = ListenerId::next();
+            tcp.listen_on(
+                listener_id,
+                format!("/ip4/0.0.0.0/tcp/{port}").parse().unwrap(),
+            )
+            .unwrap();
+            tcp.listen_on(
+                ListenerId::next(),
+                format!("/ip6/::/tcp/{port}").parse().unwrap(),
+            )
+            .unwrap();
+        }
+        #[cfg(feature = "async-io")]
+        {
+            async_std::task::block_on(async {
+                test::<async_io::Tcp>(3001);
+            })
+        }
+        let rt = ::tokio::runtime::Builder::new_current_thread()
+            .enable_io()
+            .build()
+            .unwrap();
+        rt.block_on(async {
+            test::<async_io::Tcp>(4001);
+        });
```

Comment:
Wouldn't the tokio execution also be feature flagged? In other words, does `cd transports/tcp && cargo test --feature async-io` succeed?

Reply:
no, right. Thanks for the correction Max, updated.